lib/string.js

```diff
@@ -430,4 +434,27 @@ internals.String.prototype.trim = function () {
     return obj;
 };
 
+
+internals.String.prototype.replace = function (pattern, replacement) {
+
+    Hoek.assert(pattern instanceof RegExp, 'pattern must be a RegExp');
+    Hoek.assert(typeof replacement === 'string', 'replacement must be a String');
+
+    var obj = this._test('replace', undefined, function (value, state, options) {
+
+        if (options.convert) { // we do NOT check the pattern if not converting, that's the job of "regex(...)"
+            return null;
+        }
+
+        return Errors.create('string.replace', { value: value }, state, options);
+    });
+
+    obj._flags.transform = function (value) {
```

Comment:
Use the _inner object and put that in an array, having multiple replaces is probably something people will want.